dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchFilterTranslator.cs

```diff
@@ -312,26 +317,60 @@ private void TranslateUnary(UnaryExpression unary)
                 this._filter.Append(')');
                 return;
 
+            // Handle convert over member access, for dynamic dictionary access (r => (int)r["SomeInt"] == 8)
+            case ExpressionType.Convert when this.TryBindProperty(unary.Operand, out var property) && unary.Type == property.Type:
+                this._filter.Append(property.StorageName); // TODO: Escape
+                return;
+
             default:
                 throw new NotSupportedException("Unsupported unary expression node type: " + unary.NodeType);
         }
     }
 
-    private bool TryGetField(Expression expression, [NotNullWhen(true)] out string? field)
+    private bool TryBindProperty(Expression expression, [NotNullWhen(true)] out VectorStoreRecordPropertyModel? property)
```

Comment:
This is duplicated across the connectors - we'll refactor this out to a common utility later.